app/models/mixins/vim_connect_mixin.rb

```diff
@@ -5,11 +5,9 @@ def connect(options = {})
     options[:auth_type] ||= :ws
     raise _("no credentials defined") if missing_credentials?(options[:auth_type])
 
-    options[:fault_tolerant] = true unless options.key?(:fault_tolerant)
-
-    options[:use_broker] = (self.class.respond_to?(:use_vim_broker?) ? self.class.use_vim_broker? : ManageIQ::Providers::Vmware::InfraManager.use_vim_broker?) if options[:fault_tolerant] && !options.key?(:use_broker)
-    options[:check_broker_worker] = !!options[:use_broker] unless options.key?(:check_broker_worker)
-    if options[:check_broker_worker] && !MiqVimBrokerWorker.available?
+    if !options.key?(:use_broker)
+      options[:use_broker] = MiqVimBrokerWorker.available?
```

Comment:
@Fryguy 's point here was we should always use the broker unless explicitly disabled.  That way if the broker isn't available we should throw an error instead of trying to connect directly which would be different behavior.

Reply:
Correct, if the caller directly specifies to use or not use the broker we should not overwrite that.